### ISSUE TITLE
TEMPO doesn't work correctly when the LONG press action is meant to be a HOLD

### ISSUE THREAD

**Issue by drivera73** [WhiteMagic/JoystickGremlin#342]
This is the mapping I'm trying to achieve in a TEMPO (I have it working just fine in TARGET):

- Short Press = do action X
- Long Press = repeat action Y until the button is released

Since the activation is either on press, which will always activate the short press, or on release which will only execute the long press action once, this is not currently possible.

I'm using release 13.3.

**Comment by WhiteMagic** [comment]
The problem isn't with the tempo container but rather with the macro operation mode you are using. These modes are quirky and depending on how exactly containers implement their actions will or won't work. In this case, due to how the release action gets handled the hold aspect of the macro won't work. Effectively this requires redoing how containers and actions interact.

**Comment by drivera73** [comment]
Fair enough ... is this within the cards for sometime before the next century or nah?  :)

I realize how much work this might be, just asking to get an expectation...

Failing that, can you think of how the above functionality would be possible via other means?

**Comment by WhiteMagic** [comment]
The distinction between actions and containers will disappear with R14 and everything will just be an action, i.e. nesting what are now containers inside of each other will be possible, though how far down the rabbit hole I will make sure it behaves sanely is a different question.

The side effect of this is that a lot of the actions are being reworked etc. and their interaction logic changes as well. So I will have to touch this when getting to those and it is likely that some amount of oddness will disappear in the process since there should be fewer edge cases.

**Comment by drivera-armedia** [comment]
Fair enough!! Thanks for the clarification. What is your planned/hoped timeframe for R14? Just FMI...

**Comment by WhiteMagic** [comment]
As for a timeline, honestly, I'm not even trying to predict anything. It's been much slower than I hoped for a bunch of reasons. Currently, I'm working towards something like a preview release that isn't intended to be a new version but just making sure the underlying stuff works before rebuilding the entire UI.

**Comment by drivera-armedia** [comment]
Thanks for being so forthcoming! I'll be watching closely!

**Comment by WhiteMagic** [comment]
This should now work in R14 as the action system is a bit more flexible and the tempo and similar actions have a less strict way of activating.
